### PR TITLE
Improve accuracy of metrics naming

### PR DESCRIPTION
**Pull request by MatthewScholefield**
As it is right now, when user opts in the wav file is named using python's hash function which changes between a pi and a desktop. It also uses the Device ID rather than the Account ID which makes it harder to retract user data retroactively.
 - md5 is consistent across architectures
 - Account ID is now consistent per user. Previously was using the device id

### CODE REVIEW

File: mycroft/client/speech/mic.py [hunk 1]
```diff
@@ -440,12 +443,12 @@ def _wait_until_wake_word(self, source, sec_per_buffer):
                     ww_module = self.wake_word_recognizer.__class__.__name__
 
                     ww = self.wake_word_name.replace(' ', '-')
-                    md = str(abs(hash(ww_module)))
+                    md = md5(ww_module).hexdigest()
```
[the remark this answers is not in the document]
Comment (forslund) [reply mycroft/client/speech/mic.py]: This shouldn't be an issue since it's only used for unique identification.

**Comment by forslund** [comment]
Tested and works as expected.

**Comment by forslund** [comment]
Just checked the test, This may fail if the device isn't paired?

**Comment by MatthewScholefield** [comment]
@forslund This will no longer fail when not paired. Good catch.

**Comment by MatthewScholefield** [comment]
@forslund Good to merge?

**Comment by forslund** [comment]
Yes!

**Pull request merged**